diff --git a/runner/scalesets.go b/runner/scalesets.go
index 3704f43..34deb96 100644
--- a/runner/scalesets.go
+++ b/runner/scalesets.go
@@ -95,12 +95,12 @@ func (r *Runner) DeleteScaleSetByID(ctx context.Context, scaleSetID uint) error
 
 	slog.DebugContext(ctx, "deleting scale set", "scale_set_id", scaleSet.ScaleSetID)
 	if err := scalesetCli.DeleteRunnerScaleSet(ctx, scaleSet.ScaleSetID); err != nil {
-		if !errors.Is(err, runnerErrors.ErrNotFound) {
+		if errors.Is(err, runnerErrors.ErrNotFound) {
 			slog.InfoContext(ctx, "scale set not found", "scale_set_id", scaleSet.ScaleSetID)
-			return nil
+		} else {
+			slog.With(slog.Any("error", err)).ErrorContext(ctx, "failed to delete scale set from github")
+			return fmt.Errorf("error deleting scale set from github: %w", err)
 		}
-		slog.With(slog.Any("error", err)).ErrorContext(ctx, "failed to delete scale set from github")
-		return fmt.Errorf("error deleting scale set from github: %w", err)
 	}
 	if err := r.store.DeleteScaleSetByID(ctx, scaleSetID); err != nil {
 		return fmt.Errorf("error deleting scale set: %w", err)
